mqtt/mqtt-bridge/src/persist/memory/mod.rs

```diff
@@ -0,0 +1,202 @@
+use std::{error::Error, sync::Arc};
+
+use anyhow::Result;
+use async_trait::async_trait;
+use mqtt3::proto::Publication;
+use parking_lot::Mutex;
+
+use crate::persist::{
+    memory::loader::InMemoryMessageLoader, memory::waking_map::WakingMap, Key, Persist,
+};
+
+mod loader;
+mod waking_map;
+
+/// In memory persistence implementation used for the bridge
+struct InMemoryPersist {
+    state: Arc<Mutex<WakingMap>>,
+    offset: u32,
+    loader: Arc<Mutex<InMemoryMessageLoader>>,
+}
+
+#[async_trait]
+impl<'a> Persist<'a> for InMemoryPersist {
+    type Loader = InMemoryMessageLoader;
+
+    async fn new(batch_size: usize) -> Self {
+        let state = WakingMap::new();
+        let state = Arc::new(Mutex::new(state));
+
+        let loader = InMemoryMessageLoader::new(Arc::clone(&state), batch_size).await;
+        let loader = Arc::new(Mutex::new(loader));
```

Comment:
is Mutex needed here because next is mutating the loader? It looks weird that it locks to iterate, but I don't know what is the solution.

Reply:
I had the same concern and was going to ask Denis. @dmolokanov Thoughts?